### PR TITLE
INS-20920: PostgreSQL APIv2

### PR DESCRIPTION
**Pull request by Quintero-daniel**
Enable the PostgreSQL offering on the APIv2 - Terraform provider v2

### CODE REVIEW

**Review by peter-toms — APPROVED** [review]

File: docs/resources/postgresql_cluster_v2.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+<ins>Type</ins>: string, required, immutable<br>
+<ins>Constraints</ins>: pattern: `[0-9]+\.[0-9]+\.[0-9]+`<br><br>Version of Postgresql to run on the cluster. Available versions: <ul> <li>`13.8.0`</li> <li>`14.5.0`</li> <li>`15.0.0`</li> </ul><br><br>
+*___private_network_cluster___*<br>
+<ins>Type</ins>: boolean, required, immutable<br>
+<br>Creates the cluster with private network only, see [Private Network Clusters](https://www.instaclustr.com/support/documentation/useful-information/private-network-clusters/).<br><br>
+### Input attributes - Optional
+*___two_factor_delete___*<br>
+<ins>Type</ins>: nested block, optional, updatable, see [two_factor_delete](#nested--two_factor_delete) for nested schema<br>
+<br>
+*___pci_compliance_mode___*<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:65]: Given we do not have PCI compliance do we even need this option?

File: docs/resources/postgresql_cluster_v2.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+*___intra_data_centre_replication___*<br>
+<ins>Type</ins>: nested block, required, updatable, see [intra_data_centre_replication](#nested--intra_data_centre_replication) for nested schema<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:104]: This should be immutable

File: docs/resources/postgresql_cluster_v2.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+<br>The private network address block for the Data Centre specified using CIDR address notation. The network must have a prefix length between `/12` and `/22` and must be part of a private address space.<br><br>
+### Input attributes - Optional
+*___azure_settings___*<br>
+<ins>Type</ins>: nested block, optional, immutable, see [azure_settings](#nested--azure_settings) for nested schema<br>
+<br>Azure specific settings for the Data Centre. Cannot be provided with AWS or GCP settings.<br><br>
+*___gcp_settings___*<br>
+<ins>Type</ins>: nested block, optional, immutable, see [gcp_settings](#nested--gcp_settings) for nested schema<br>
+<br>GCP specific settings for the Data Centre. Cannot be provided with AWS or Azure settings.<br><br>
+*___tag___*<br>
+<ins>Type</ins>: repeatable nested block, optional, immutable, see [tag](#nested--tag) for nested schema<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:117]: why is this immutable? I thought we can do tags afterwards

File: docs/resources/postgresql_cluster_v2.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+<br>List of tags to apply to the Data Centre. Tags are metadata labels which  allow you to identify, categorize and filter clusters. This can be useful for grouping together clusters into applications, environments, or any category that you require.<br><br>
+*___inter_data_centre_replication___*<br>
+<ins>Type</ins>: nested block, optional, updatable, see [inter_data_centre_replication](#nested--inter_data_centre_replication) for nested schema<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:120]: again immutable

File: docs/resources/postgresql_cluster_v2.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+<br>
+*___pg_bouncer___*<br>
+<ins>Type</ins>: nested block, optional, updatable, see [pg_bouncer](#nested--pg_bouncer) for nested schema<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:123]: Immutable

File: docs/resources/postgresql_cluster_v2.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+<ins>Constraints</ins>: allowed values: [ `CASSANDRA`, `SPARK_MASTER`, `SPARK_JOBSERVER`, `KAFKA_BROKER`, `KAFKA_DEDICATED_ZOOKEEPER`, `KAFKA_ZOOKEEPER`, `KAFKA_SCHEMA_REGISTRY`, `KAFKA_REST_PROXY`, `APACHE_ZOOKEEPER`, `POSTGRESQL`, `PGBOUNCER`, `KAFKA_CONNECT`, `KAFKA_KARAPACE_SCHEMA_REGISTRY`, `KAFKA_KARAPACE_REST_PROXY`, `CADENCE`, `MONGODB`, `REDIS_MASTER`, `REDIS_REPLICA`, `OPENSEARCH_DASHBOARDS`, `OPENSEARCH_COORDINATOR`, `OPENSEARCH_MASTER`, `OPENSEARCH_DATA_AND_INGEST` ]<br><br>The roles or purposes of the node. Useful for filtering for nodes that have a specific role.<br><br>
+*___public_address___*<br>
+<ins>Type</ins>: string, read-only<br>
+<br>Public IP address of the node.<br><br>
+<a id="nested--inter_data_centre_replication"></a>
+## Nested schema for `inter_data_centre_replication`
+
+### Input attributes - Required
+*___is_primary_data_centre___*<br>
+<ins>Type</ins>: boolean, required, updatable<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:195]: this should be immutable for customers and might be updated by us

File: docs/resources/postgresql_cluster_v2.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+<br>Is this Data centre considered to be the primary (only required if multiple data centres defined).<br><br>
+<a id="nested--pg_bouncer"></a>
+## Nested schema for `pg_bouncer`
+
+### Input attributes - Required
+*___pool_mode___*<br>
+<ins>Type</ins>: string, required, immutable<br>
+<ins>Constraints</ins>: allowed values: [ `TRANSACTION`, `SESSION`, `STATEMENT` ]<br><br>Creates PgBouncer with the selected mode, see [PgBouncer pool modes] (https://www.instaclustr.com/support/documentation/postgresql-add-ons/using-pgbouncer/)<br><br>
+*___pg_bouncer_version___*<br>
+<ins>Type</ins>: string, required, updatable<br>
```
Comment (tharinduamila-insta) [inline docs/resources/postgresql_cluster_v2.md:205]: immutable?

**Review by tharinduamila-insta — CHANGES_REQUESTED** [review]
there are many more attributes that need immutable set to them

**Review by tharinduamila-insta — APPROVED** [review]

**Review by arjunrajnetapp — APPROVED** [review]

**Pull request merged**